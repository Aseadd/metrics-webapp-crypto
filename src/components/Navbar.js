import React from 'react';
import PropTypes from 'prop-types';
import { MdMic, MdSettings } from 'react-icons/md';
import { NavLink } from 'react-router-dom';
import './App.css';

const Navbar = (props) => {
  const { Title } = props;

  return (
    <div
      className="d-flex justify-content-between align-items-center navbar"
      data-testid="navbarTest"
    >
      <NavLink to="/" exact style={{ textDecoration: 'none' }}>
        <h1>{Title}</h1>
      </NavLink>
      <div className="fs-4">
        <MdMic className="me-3" />
        <MdSettings />
      </div>
    </div>
  );
};

Navbar.propTypes = {
  Title: PropTypes.string.isRequired,
};
export default Navbar;
