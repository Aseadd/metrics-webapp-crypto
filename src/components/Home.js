import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import CryptoCard from './card';
import TotalCard from './totalCard';
import crypto from '../assets/bitcoin-logo.png';
import Navbar from './Navbar';
import './App.css';

function Home() {
  const coins = useSelector((state) => state.coinsReducer.coins);
  const [name, setName] = useState('');
  const handleSearch = (e) => {
    setName(e.target.value);
  };
  const filteredCoins = coins
    ? coins.filter(
      (coin) => coin.name
        .toLowerCase()
        .includes(
          name.toLowerCase()
                || coin.symbol.toLowerCase().includes(name.toLowerCase()),
        ) || coin.rank.toString().includes(name.toString()),
    )
    : [];
  if (coins === undefined) {
    return (
      <div className="d-flex justify-content-center align-items-center ring">
        <h1>Loading...</h1>
      </div>
    );
  }
  return (
    <div>
      <Navbar Title="Crypto Tracker" />

      <div className="container">
        <div className="d-flex align-items-center justify-content-around total-card-container">
          <div className="d-flex align-items-center justify-content-around total-card">
            <img src={crypto} alt="Crypto Logo" className="logo-img" />
            <TotalCard header={coins} />
          </div>
          <div className="coinImg">
            <div className="coinIcons">
              <p>ðŸ’°</p>
              <p>â‚¿ </p>
              <p>ðŸª™</p>
            </div>
          </div>
          <div className="d-flex justify-content-center align-items-center search-input">
            <input
              type="text"
              placeholder="Search Crypto by name, symbol or rank"
              className="search"
              onChange={handleSearch}
              value={name}
            />
          </div>
        </div>
        <div className="coins">
          {filteredCoins?.map((coin) => (
            <CryptoCard key={coin.id} coin={coin} />
          ))}
        </div>
      </div>
    </div>
  );
}

export default Home;
