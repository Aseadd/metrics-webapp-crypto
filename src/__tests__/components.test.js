import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import Home from '../components/Home';
import Navbar from '../components/Navbar';

describe('Test the Components', () => {
  test('Renders the Navbar component ', () => {
    render(
      <Provider store={store}>
        <Router>
          <Navbar />
        </Router>
      </Provider>,
    );
    expect(screen.getByTestId('navbarTest')).toBeInTheDocument();
  });

  test('Renders the Home component ', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Home />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
