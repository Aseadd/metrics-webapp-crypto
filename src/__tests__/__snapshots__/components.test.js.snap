// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Components Renders the Home component  1`] = `
<div
  className="d-flex justify-content-center align-items-center ring"
>
  <h1>
    Loading...
  </h1>
</div>
`;
